### DISCUSSION TITLE
Mime Type for .stl (Stereolithography Mesh) File

### DISCUSSION THREAD

**Discussion by BobSnrub** [pocketbase/pocketbase#450] [Q&A]
I'm currently trying to allow the upload of .zip and .stl files.  I was able to get the .zip files to upload but when trying to upload .stl files it will tell me that the mime type is not allow.  These are the two mime types I have tried for .stl: `model/x.stl-binary,application/sla`.  Anyone have any suggestions?

![image](https://user-images.githubusercontent.com/56611572/190269448-21c0ac76-6c85-4b11-bfee-763627129e53.png)

**Comment by ganigeorgiev** [comment]
The upstream library that we use for mimetype detection ([`gabriel-vasile/mimetype`](https://github.com/gabriel-vasile/mimetype)) doesn't have mapped the `application/sla` mimetype byte signature in the package. You can open a PR in the upstream library (you may need to provide a sample file).

For full list of the currently supported mimetypes and their aliases, you could check https://github.com/gabriel-vasile/mimetype/blob/master/supported_mimes.md.

An alternative is to find other mimetype that has similar signature and use that instead. For example ms office files has the same signature as zip (at least the first couple of bytes).

**Reply by BobSnrub** [reply]
Ok, thanks for bringing this to my attention.  I see that someone else has opened a pull request in that upstream library for severeral 3D file formats including .stl.
